src/mscorlib/src/System/Char.cs

```diff
@@ -929,10 +927,10 @@ public static String ConvertFromUtf32(int utf32)
             }
             // This is a sumplementary character.  Convert it to a surrogate pair in UTF-16.
             utf32 -= UNICODE_PLANE01_START;
-            char[] surrogate = new char[2];
-            surrogate[0] = (char)((utf32 / 0x400) + (int)CharUnicodeInfo.HIGH_SURROGATE_START);
-            surrogate[1] = (char)((utf32 % 0x400) + (int)CharUnicodeInfo.LOW_SURROGATE_START);
-            return (new String(surrogate));
+            char* surrogate = stackalloc char[2];
+            surrogate[0] = (char)((utf32 / 0x400) + HIGH_SURROGATE_START);
+            surrogate[1] = (char)((utf32 % 0x400) + LOW_SURROGATE_START);
+            return new string(surrogate, 0, 2);
```

Comment:
I'm not sure what may be more performant, stackallocing a char* and constructing a string, or the following that uses an internal method I've seen in System.Globalization and StringBuilder:

```
string result = string.FastAllocateString(2);
fixed(char* pResult = result)
{
     pResult[0] = (char)((utf32 / 0x400) + HIGH_SURROGATE_START);
     pResult[1] = (char)((utf32 % 0x400) + LOW_SURROGATE_START);
}
```
@ellismg let me know what you think